### PR TITLE
build.make: allow repos to use 'go mod' for vendoring

### PR DESCRIPTION
**Pull request by pohly**
How a repo does vendoring is detected based on the presence of
Gopkg.toml.

The vendor check with `dep` was all done locally, but the
corresponding check for `go mod` requires network access. The check
therefore gets skipped when running in the Prow CI in situations where
we are sure that it isn't needed (for example, in a periodic job).

This is required for https://github.com/kubernetes-csi/csi-test/pull/188

```release-note
allow vendoring with 'go mod'
```
/kind feature

### CODE REVIEW

File: build.make [hunk 1]
```diff
@@ -118,14 +118,39 @@ test-fmt:
...
+	  else \
+		echo "Repo uses 'go mod' for vendoring."; \
+		if [ "$${JOB_NAME}" ] && \
+                   ( [ "$${JOB_TYPE}" != "presubmit" ] || \
+                     [ $$(git diff "${PULL_BASE_SHA}..HEAD" -- go.mod go.sum vendor release-tools | wc -l) -eq 0 ] ); then \
+			echo "Skipping vendor check because the Prow pre-submit job does not change vendoring."; \
+		elif ! GO111MODULE=on go mod vendor; then \
+			echo "ERROR: vendor check failed."; \
+			false; \
+		elif [ $$(git status --porcelain=2 -- vendor | wc -l) -gt 0 ]; then \
```
Comment (hoyho) [inline build.make]: may get an error when execute `git status --porcelain=2 -- vendor | wc -l`  with different git version?

```bash
 [root@vm43 csi-test]# git status --porcelain=2 -- vendor | wc -l
error: option `porcelain' takes no value
usage: git status [options] [--] <pathspec>...

    -v, --verbose         be verbose
    -s, --short           show status concisely
    -b, --branch          show branch information
    --porcelain           machine-readable output
    --long                show status in long format (default)
    -z, --null            terminate entries with NUL
    -u, --untracked-files[=<mode>]
                          show untracked files, optional modes: all, normal, no. (Default: all)
    --ignored             show ignored files
    --ignore-submodules[=<when>]
                          ignore changes to submodules, optional when: all, dirty, untracked. (Default: all)
    --column[=<style>]    list untracked files in columns

0
```

Comment (pohly) [reply build.make]: Which version of git are you using?

We can drop the `=2`, the original format should be good enough. All we care for is "empty" vs. "not empty" anyway.

Comment (pohly) [reply build.make]: Thanks for testing, BTW.

Comment (pohly) [reply build.make]: I've removed the `=2`.

**Pull request merged**